build/configuration-properties-collector/src/main/java/org/hibernate/search/configuration/properties/collector/impl/ConfigurationPropertyCollector.java

```diff
@@ -0,0 +1,219 @@
+		if ( annotation.flatMap( a -> a.attribute( "ignore", boolean.class ) ).orElse( false ) ) {
+			return;
+		}
+
+		ConfigurationProperty.Key key = extractKey(
+				constant,
+				classPrefix,
+				annotation.flatMap( a -> a.multiAttribute( "prefix", String.class ) )
+		);
+		if ( !key.matches( ignoreKeys ) ) {
+			// Try to find a default value. Assumption is that the settings class has an inner class called "Defaults" and
+			// the key for the default value is exactly the same as the config constant name:
+			Object value = findDefault( constant );
+
+			properties.put(
+					constant.getEnclosingElement().toString() + "#" + constant.getSimpleName().toString(),
+					new ConfigurationProperty()
+							.javadoc( extractJavadoc( constant ) )
+							.key( key )
+							.sourceClass( constant.getEnclosingElement().toString() )
+							.type( extractType( constant ) )
+							.defaultValue( value )
+			);
+		}
+	}
+
+
+	private ConfigurationProperty.Key extractKey(VariableElement constant, Optional<List<String>> classPrefix, Optional<List<String>> constantPrefix) {
+		List<String> prefix;
+		if ( constantPrefix.isPresent() ) {
+			prefix = constantPrefix.get();
+		}
+		else if ( classPrefix.isPresent() ) {
+			prefix = classPrefix.get();
+		}
+		else {
+			prefix = configPrefix;
+		}
+
+		return new ConfigurationProperty.Key(
+				prefix,
+				Objects.toString( constant.getConstantValue(), "NOT_FOUND#" + constant.getSimpleName() )
+		);
+	}
+
+	private HibernateSearchConfiguration.Type extractType(VariableElement constant) {
+		String packageName = packageElement( constant ).getQualifiedName().toString();
+		return SPI_PATTERN.matcher( packageName ).matches() ?
+				HibernateSearchConfiguration.Type.SPI :
+				HibernateSearchConfiguration.Type.API;
+	}
+
+	private String extractJavadoc(VariableElement constant) {
+		try {
+			Element enclosingClass = constant.getEnclosingElement();
+			Path docs = javadocsLocation.resolve(
+					enclosingClass.toString().replace( ".", File.separator ) + ".html"
+			);
+
+			// calling getEnclosingElement() on class should return a package:
+			String packagePath = packageElement( enclosingClass ).getQualifiedName().toString().replace( ".", File.separator );
+
+			Document javadoc = Jsoup.parse( docs.toFile() );
+
+			org.jsoup.select.Elements blocks = javadoc.select( "#" + constant.getSimpleName() + " .block" );
+			if ( !blocks.isEmpty() ) {
+				org.jsoup.nodes.Element block = blocks.get( 0 );
+				for ( org.jsoup.nodes.Element link : block.getElementsByTag( "a" ) ) {
+					String href = link.attr( "href" );
+					// only update links if they are not external:
+					if ( !link.hasClass( "external-link" ) ) {
+						if ( href.startsWith( "#" ) ) {
+							href = enclosingClass.getSimpleName().toString() + ".html" + href;
+						}
+						href = javadocsBaseLink + packagePath + "/" + href;
+					}
```

Comment:
Real question, not necessarily asking you to do it: would it be complex to replace links to other configuration property constants with internal links to the Asciidoc you're currently generating?

E.g. in the example below, 
`"client.instance"` would point to the item about `hibernate.search.backend.client.instance` in the reference documentation, instead of pointing to the javadoc.

![image](https://user-images.githubusercontent.com/412878/212905220-7117e0bc-3a09-496d-9a6b-f25f71a955a2.png)

Reply:
would need to see how anchor links are generated by asciidoctor... Right now we pass this  HTML as is to ascidoctor and it just adds it without processing. So if we can predict the link, we can replace it. Otherwise, we could delay the replacement up until asciidocwriter and then try to play with formatting there and use the ascidoctor anchor/reference formatting.
I'll see what is available to us within this HTML and will let you know.